src/cirrus/lib/statedb.py

```diff
@@ -440,7 +451,11 @@ def query(self, collections_workflow: str, state: str=None, since: str=None,
             kwargs['ExclusiveStartKey'] = {k: kwargs['ExclusiveStartKey'][k] for k in keys}
 
         if index:
-            resp = self.table.query(IndexName=index, KeyConditionExpression=expr, Select=select, ScanIndexForward=sort_ascending, **kwargs)
+            if filter_exp:
+                resp = self.table.query(IndexName=index, KeyConditionExpression=expr, FilterExpression=filter_exp, Select=select, 
+                                        ScanIndexForward=sort_ascending, **kwargs)
+            else:
+                resp = self.table.query(IndexName=index, KeyConditionExpression=expr, Select=select, ScanIndexForward=sort_ascending, **kwargs)
         else:
             resp = self.table.query(KeyConditionExpression=expr, Select=select,  ScanIndexForward=sort_ascending, **kwargs)
```

Comment:
Same feedback as above: we should have just one call to `self.table.query` and build the keyword args up conditionally.